source/core/TDCodecVia.cpp

```diff
@@ -1058,12 +1059,17 @@ Int32 TDViaParser::ParseData(TypeRef type, void* pData)
         case kEncoding_UInt:
         case kEncoding_S2CInt:
             {
+                Boolean is64Bit = type->IsA(&TypeCommon::TypeInt64) || type->IsA(&TypeCommon::TypeUInt64);
                 IntMax value = 0;
                 Boolean overflow = false;
                 Utf8Char sign = 0;
                 _string.EatWhiteSpaces();
+                if (_options._bQuote64BitNumbers && is64Bit) {
+                    _string.EatChar('"');
```

Comment:
When reading JSON the parser should take a JSON number and turn it to the Vireo numeric or a JSON string that represents a number.

Make sure for 64 Int and Unint coming from JSON it can take both the json:

```json
18446744073709551615
```

and 

```json
"18446744073709551615"
```

Reply:
You may need to save the result of EatChar to use below